Quadriga/src/main/java/edu/asu/spring/quadriga/web/network/NetworkListController.java

```diff
@@ -15,87 +15,150 @@
+     */
+    @RequestMapping(value = "auth/networks", method = RequestMethod.GET)
     public String listNetworks(ModelMap model, Principal principal) throws QuadrigaStorageException, JSONException {
-		IUser user = userManager.getUser(principal.getName());
-		List<INetwork> networkList=networkManager.getNetworkList(user);
-		model.addAttribute("userId", user.getUserName());
-		model.addAttribute("networkList", networkList);
-		return "auth/networks";
-	}
-
-
-	/**
-	 * Get the network displayed on to JSP by passing JSON string
-	 * @author Lohith Dwaraka
-	 * @param networkId
-	 * @param model
-	 * @param principal
-	 * @return
-	 * @throws QuadrigaStorageException
-	 * @throws JAXBException
-	 */
-	@RequestMapping(value = "auth/networks/visualize/{networkId}", method = RequestMethod.GET)
-	public String visualizeNetworks(@PathVariable("networkId") String networkId, ModelMap model, Principal principal) throws QuadrigaStorageException, JAXBException {
-		INetwork network = networkManager.getNetwork(networkId);
-		if(network==null){
-			return "auth/404";
-		}
-		ITransformedNetwork transformedNetwork= transformationManager.getTransformedNetwork(networkId);
-		
-		String nwId = "\""+networkId+"\"";
-		model.addAttribute("networkid",nwId);
-		model.addAttribute("network", network);
-		
-		String json = null;
-		if(transformedNetwork!=null){
-			json = d3Creator.getD3JSON(transformedNetwork.getNodes(), transformedNetwork.getLinks());
-		}
-		model.addAttribute("jsonstring",json);
-
-		return "auth/networks/visualize";
-	}
+        IUser user = userManager.getUser(principal.getName());
+        List<INetwork> networkList = networkManager.getNetworkList(user);
+        model.addAttribute("userId", user.getUserName());
+        model.addAttribute("networkList", networkList);
+        return "auth/networks";
+    }
+
+    /**
+     * Get the network displayed on to JSP by passing JSON string
+     * 
+     * @author Lohith Dwaraka, Chiraag Subramanian
+     * @param networkId
+     * @param model
+     * @param principal
+     * @return
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @RequestMapping(value = "auth/networks/visualize/{networkId}", method = RequestMethod.GET)
+    public String visualizeNetworks(@PathVariable("networkId") String networkId, ModelMap model, Principal principal)
+            throws QuadrigaStorageException, JAXBException {
+        // Identify the User
+        String userId = principal.getName();
+
+        // Get network details using the networkId
+        INetwork network = networkManager.getNetwork(networkId);
+
+        // Get details of workspace associated with the network using
+        // network-workspace mapping details
+        IWorkSpace workspace = network.getNetworkWorkspace().getWorkspace();
+
+        // Get projectId of the project to which the workspace belongs
+        String projectId = workspaceManager.getProjectIdFromWorkspaceId(workspace.getWorkspaceId());
```

Comment:
the workspace should have the project id, no need for another database call here

Reply:
IWorkSpace has a method : getProject() to retrieve the Project associated with the workspace. However we get null when getProject() is invoked on the Workspace. Hence, the need for the additional database access to fetch the projectId